taichi/transforms/binary_op_simplify.cpp

```diff
@@ -0,0 +1,94 @@
+#include "taichi/ir/ir.h"
+#include "taichi/ir/transforms.h"
+#include "taichi/ir/visitors.h"
+#include "taichi/program/program.h"
+
+TLANG_NAMESPACE_BEGIN
+
+class BinaryOpSimp : public BasicStmtVisitor {
+ public:
+  using BasicStmtVisitor::visit;
+  bool fast_math;
+  DelayedIRModifier modifier;
+
+  explicit BinaryOpSimp(bool fast_math_)
+      : BasicStmtVisitor(), fast_math(fast_math_) {
+  }
+
+  void visit(BinaryOpStmt *stmt) override {
+    // swap lhs and rhs if lhs is a const and op is commutative
+    auto const_lhs = stmt->lhs->cast<ConstStmt>();
+    if (const_lhs && is_commutative(stmt->op_type)) {
```

Comment:
I think we don't need to swap if rhs is already a ConstStmt -- we can't assume constant_fold is done before this pass.
```suggestion
    if (const_lhs && is_commutative(stmt->op_type) && !stmt->rhs->is<ConstStmt>()) {
```